salt/cloud/clouds/libvirt.py

```diff
@@ -1159,6 +1159,8 @@ def start(name, call=None):
             found.append(domain)
         except libvirtError:
             pass
+        finally:
+            conn.close()
```

Comment:
Sorry for not having been precise enough. I think closing here may be too early: I would bet the domain objects stored in `found` array would then be freed.

There are two possible solutions:
1. Store the connection handles in a list and close them once all the domains have been started (or any other action for the similar functions)
2. Close the connections that gave no result within the `for` loop and the others with something like:
```python
for dom in found:
    dom.connect().close()
```

Reply:
Hello, I will look into that :)
I'm not the original author for these start, stop and reboot functions, so I need to talk with my colleague about what behaviour is expected exactly.

From what I understand, the functions are targeting a single VM, not a list.
To locate where the VM is, we connect to each provider in a loop.
If we find zero or more than one VM (e.g. there are VM with the same name on different hypervisor), we do nothing.

The functions are working in their current states, not sure how. I agree that closing the connection before calling start() should fail, but it does not.

Reply:
OK, I see better how it works now. The real connection `close()` is done after the start so we are safe